presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -466,6 +486,61 @@ private RelationPlan createTableWriterPlan(
         return new RelationPlan(commitNode, analysis.getRootScope(), commitNode.getOutputSymbols());
     }
 
+    /*
+    According to the standard, for the purpose of store assignment (INSERT),
+    no non-space characters of a character string, and no non-zero octets
+    of a binary string must be lost when the inserted value is truncated to
+    fit in the target column type.
+    The following method returns a cast from source type to target type
+    with a guarantee of no illegal truncation.
+    TODO Once BINARY and parametric VARBINARY types are supported, they should be handled here.
+    TODO This workaround is insufficient to handle structural types
+     */
+    private Expression noTruncationCast(Expression fromExpression, Type fromType, Type toType)
```

Comment:
I'd just call the expression argument `Expression`

Reply:
I think that truncating trailing spaces is not a big deal to either of the regexp implementations.
The `functionRegistry` is initialised with the regexp implementation declared in the `FeaturesConfig`. It can only resolve `regexReplace(VARCHAR, regexType)` if the `regexType` matches.
If we want to pick one particular regexp implementation regardless of the config, we might need to add the corresponding function to the functionRegistry. That could affect other sites where `regexp_replace` function is resolved.